client/cli/src/logging.rs

```diff
@@ -0,0 +1,336 @@
+				Level::DEBUG => f.pad(DEBUG_STR),
+				Level::INFO => f.pad(INFO_STR),
+				Level::WARN => f.pad(WARN_STR),
+				Level::ERROR => f.pad(ERROR_STR),
+			}
+		}
+	}
+}
+
+struct FmtThreadName<'a> {
+	name: &'a str,
+}
+
+impl<'a> FmtThreadName<'a> {
+	pub(crate) fn new(name: &'a str) -> Self {
+		Self { name }
+	}
+}
+
+// NOTE: the following code has been duplicated from tracing-subscriber
+//
+//       https://github.com/tokio-rs/tracing/blob/2f59b32/tracing-subscriber/src/fmt/format/mod.rs#L845
+impl<'a> fmt::Display for FmtThreadName<'a> {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		use std::sync::atomic::{
+			AtomicUsize,
+			Ordering::{AcqRel, Acquire, Relaxed},
+		};
+
+		// Track the longest thread name length we've seen so far in an atomic,
+		// so that it can be updated by any thread.
+		static MAX_LEN: AtomicUsize = AtomicUsize::new(0);
+		let len = self.name.len();
+		// Snapshot the current max thread name length.
+		let mut max_len = MAX_LEN.load(Relaxed);
+
+		while len > max_len {
+			// Try to set a new max length, if it is still the value we took a
+			// snapshot of.
+			match MAX_LEN.compare_exchange(max_len, len, AcqRel, Acquire) {
+				// We successfully set the new max value
+				Ok(_) => break,
+				// Another thread set a new max value since we last observed
+				// it! It's possible that the new length is actually longer than
+				// ours, so we'll loop again and check whether our length is
+				// still the longest. If not, we'll just use the newer value.
+				Err(actual) => max_len = actual,
+			}
+		}
+
+		// pad thread name using `max_len`
+		write!(f, "{:>width$}", self.name, width = max_len)
+	}
+}
+
+struct FmtCtx<'a, S, N> {
+	ctx: &'a FmtContext<'a, S, N>,
+	span: Option<&'a span::Id>,
+	ansi: bool,
+}
+
+impl<'a, S, N: 'a> FmtCtx<'a, S, N>
+where
+	S: Subscriber + for<'lookup> LookupSpan<'lookup>,
+	N: for<'writer> FormatFields<'writer> + 'static,
+{
+	pub(crate) fn new(
+		ctx: &'a FmtContext<'_, S, N>,
+		span: Option<&'a span::Id>,
+		ansi: bool,
+	) -> Self {
+		Self { ctx, ansi, span }
+	}
+
+	fn bold(&self) -> Style {
+		if self.ansi {
+			return Style::new().bold();
+		}
+
+		Style::new()
+	}
+}
+
+// NOTE: the following code has been inspired from tracing-subscriber
+//
+//       https://github.com/tokio-rs/tracing/blob/2f59b32/tracing-subscriber/src/fmt/format/mod.rs#L711
+impl<'a, S, N: 'a> fmt::Display for FmtCtx<'a, S, N>
+where
+	S: Subscriber + for<'lookup> LookupSpan<'lookup>,
+	N: for<'writer> FormatFields<'writer> + 'static,
+{
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		let bold = self.bold();
+		let mut seen = false;
+
+		let span = self
+			.span
+			.and_then(|id| self.ctx.span(&id))
+			.or_else(|| self.ctx.lookup_current());
+
+		let scope = span
+			.into_iter()
+			.flat_map(|span| span.from_root().chain(iter::once(span)));
+
+		for name in scope
+			.map(|span| span.metadata().name())
+			.filter(|&x| x != "substrate-node")
```

Comment:
might be worth having a `const` for `"substrate-node"`

Reply:
It would be a pain because this const is in 2 crates: sc-cli and sc-cli-derive